matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/pushers/PushersService.kt

```diff
@@ -80,9 +97,10 @@ interface PushersService {
                          eventId: String)
 
     /**
-     * Remove the http pusher
+     * Remove the pusher
+     * When the appId is m.email then an email address is expected in the pushkey
      */
-    suspend fun removeHttpPusher(pushkey: String, appId: String)
+    suspend fun removePusher(pushkey: String, appId: String)
```

Comment:
renamed because _technically_ this function can remove email and http pushers

Reply:
I originally had that but it felt odd that `PushGateway` could still use the `removeHttpPusher` for both http and email pushers

the `PushersManager` has symmetrical register/unregister for http and email but that's a layer above the SDK

we could force the different flows based on appId

```kotlin
fun removePusher(pusher: Pusher) {
  when (pusher.appId) {
      Pusher.APP_ID_EMAIL -> session.removeEmailPusher(pusher.pushKey)
      else -> session.removeHttpPusher(pusher.pushKey, pusher.appId)
  }
}
``` 

what do you think?

Reply:
The `PushersManager` is on the app side, so I would say its Api is less "important" than the SDK API `PushersService`. I agree that `removeHttpPusher` is not KO to remove email pushers.

Reply:
after speaking offline, I've opted for adding a 3rd method for removing `Pushers` 4482cbdaa649ff299044a064fa1a9957a3db1819 this has the benefit of avoiding redundancy in checking the appId if we already have a `pusher` instance and we can have dedicated symmetrical http/email removal functions for convenience